devito/finite_differences/derivative.py

```diff
@@ -192,6 +193,6 @@ def evaluate(self):
                                    matvec=self.transpose, stagger=self.stagger)
         else:
             res = generic_derivative(expr, *self.dims, self.fd_order,
-                                     self.deriv_order, stagger=self.stagger,
+                                     self.deriv_order, stagger=self.stagger[0],
```

Comment:
why this change?

Reply:
Single dimension so stagger is (stagger[0],) and need to pass it not as tuple to generic_derivative. Ignores the stagger setup without it

Reply:
what do you mean "by single dimension", and how do you know?

it feels really weird that you're passing ``*self.dims`` (ie all dimensions), but only one stagger value

Reply:
stagger is a kwargs so can't pass *stagger.

>and how do you know?
because line above treats `len(dim)>1` this case is only for single dimension.

Reply:
ah ok... I was expecting ``self.dims[0]``... that ``*self.dims`` confused me at first